lib/src/model/keychain.dart

```diff
@@ -209,6 +210,42 @@ class Keychain with _$Keychain {
       'verificationMethod': verificationMethods,
     };
   }
+
+  ({Uint8List secret, List<AuthorizedKey> authorizedPublicKeys})
+      ecEncryptServiceSeed(
+    String service,
+    List<String> publicKeys, {
+    String pathSuffix = '',
+  }) {
+    if (services[service] == null) {
+      throw Exception(
+        "Service doesn't exist in the keychain",
+      );
+    }
+
+    final serviceSelected = services[service];
```

Comment:
Here you could throw an error if the services is using a derivation path without index.
This function will mainly be used to encrypt the seed for a smart contract, so the service must use a derivation path without index to match the classical derivation done by the nodes

Reply:
ok. but why don't you do the same thing in libjs ?

Reply:
This is done in libjs:
https://github.com/archethic-foundation/libjs/blob/1d399b851e0951c4197a778238f453a89d652950/src/keychain.ts#L302